packages/dev-server/src/middleware/clientLogsMiddleware.ts

```diff
@@ -0,0 +1,93 @@
+import http from 'http';
+import Log from '@expo/bunyan';
+
+type ConsoleLogLevel = 'info' | 'warn' | 'error' | 'debug';
+
+export default function clientLogsMiddleware(logger: Log) {
+  return function(
+    req: http.IncomingMessage & { body?: any },
+    res: http.ServerResponse,
+    next: (err?: any) => void
+  ) {
+    try {
+      const deviceId = req.headers['device-id'];
+      const deviceName = req.headers['device-name'];
+      if (!deviceId) {
+        res.writeHead(400).end('Missing Device-Id.');
+        return;
+      }
+      if (!deviceName) {
+        res.writeHead(400).end('Missing Device-Name.');
+        return;
+      }
+      if (!req.body) {
+        res.writeHead(400).end('Missing request body.');
+        return;
+      }
+      handleDeviceLogs(logger, deviceId.toString(), deviceName.toString(), req.body);
+    } catch (error) {
+      logger.error({ tag: 'expo' }, `Error getting device logs: ${error} ${error.stack}`);
+      next(error);
+    }
+    res.end('Success');
+  };
+}
+
+function isIgnorableBugReportingExtraData(body: any[]) {
+  return body.length === 2 && body[0] === 'BugReporting extraData:';
+}
+
+function isAppRegistryStartupMessage(body: any[]) {
+  return (
+    body.length === 1 &&
+    (/^Running application "main" with appParams:/.test(body[0]) ||
+      /^Running "main" with \{/.test(body[0]))
+  );
+}
+
+function handleDeviceLogs(logger: Log, deviceId: string, deviceName: string, logs: any) {
+  for (let i = 0; i < logs.length; i++) {
+    const log = logs[i];
+    let body = typeof log.body === 'string' ? [log.body] : log.body;
```

Comment:
```suggestion
    let body = typeof log.body === 'string' ? [log.body] : log.body;
```
Is it always `string | string[]`? Might want to change this to `Array.isArray()`

Reply:
I believe in practice it's always `LogData[]`, which means it's never anything else than an array and this check might be unnecessary (https://github.com/expo/expo/blob/e8ae25a7c9646adbb5641d0a792542a6d351e88e/packages/expo/src/logs/RemoteLogging.ts#L10-L25). Changed this to `isArray` anyway, as that's a more robust way to check this.